src/DurableTask.AzureStorage/Partitioning/AppLeaseManager.cs

```diff
@@ -0,0 +1,329 @@
+                        this.taskHub,
+                        this.workerName,
+                        this.taskHubContainerName,
+                        this.appLeaseId,
+                        ex.Message);
+
+                    this.settings.Logger.PartitionManagerError(
+                        this.accountName,
+                        this.taskHub,
+                        this.workerName,
+                        this.taskHubContainerName,
+                        $"AppLeaseManager failed to renew lease. AppLeaseId: {this.appLeaseId} Exception: {ex}");
+                }
+                else
+                {
+                    // Eat any exceptions during renew and keep going.
+                    // Consider the lease as renewed.  Maybe lease store outage is causing the lease to not get renewed.
+                    renewed = true;
+                }
+            }
+            finally
+            {
+                this.stats.StorageRequests.Increment();
+            }
+
+            this.settings.Logger.LeaseRenewalResult(
+                this.accountName,
+                this.taskHub,
+                this.workerName,
+                this.taskHubContainerName,
+                renewed,
+                this.appLeaseId,
+                errorMessage);
+
+            return renewed;
+        }
+
+        private async Task ReleaseLeaseAsync()
+        {
+            try
+            {
+                AccessCondition accessCondition = new AccessCondition() { LeaseId = this.appLeaseId };
+                await this.taskHubContainer.ReleaseLeaseAsync(accessCondition);
+
+                this.isLeaseOwner = false;
+
+                this.settings.Logger.LeaseRemoved(
+                    this.accountName,
+                    this.taskHub,
+                    this.workerName,
+                    this.taskHubContainerName,
+                    this.appLeaseId);
+            }
+            catch (Exception)
+            {
+                this.settings.Logger.LeaseRemovalFailed(
+                    this.accountName, 
+                    this.taskHub, 
+                    this.workerName,
+                    this.taskHubContainerName, 
+                    this.appLeaseId);
+            }
+        }
+
+        public async Task DeleteContainerAsync()
+        {
+            try
+            {
+                if (this.isLeaseOwner)
+                {
+                    AccessCondition accessCondition = new AccessCondition() { LeaseId = appLeaseId };
+                    await this.taskHubContainer.DeleteIfExistsAsync(accessCondition, null, null);
+                }
+                else
+                {
+                    await this.taskHubContainer.DeleteIfExistsAsync();
+                }
+            }
+            catch (StorageException ex)
+            { 
+                if (ex.RequestInformation.HttpStatusCode != 404)
+                {
+                    ExceptionDispatchInfo.Capture(ex).Throw();
```

Comment:
You can just use the `throw` statement by itself. No need for using `ExceptionDispatchInfo`. That's more for cases where you need to rethrow an exception outside of a catch block.
```suggestion
                    throw;
```